@here/harp-mapview/lib/MapView.ts

```diff
@@ -1791,6 +1839,27 @@ export class MapView extends THREE.EventDispatcher {
         this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
     }
 
+    private detectCurrentFps(now: number) {
+        if (this.m_previousRequestAnimationTime === undefined) {
+            this.m_previousRequestAnimationTime = now;
+            return;
+        }
+
+        // Skip the first frames, they are from not originated from requestAnimationFrame()
+        if (this.m_frameNumber > 5) {
+            const currentFps = 1000 / (now - this.m_previousRequestAnimationTime);
+            this.m_frameTimeRing[this.m_frameTimeIndex] = currentFps;
+            this.m_frameTimeIndex = (this.m_frameTimeIndex + 1) % FRAME_RATE_RING_SIZE;
+
+            let sum: number = 0;
+            for (let i = 0; i < FRAME_RATE_RING_SIZE; i++) {
```

Comment:
minor, you could save some code by using reduce(): https://www.w3schools.com/jsref/jsref_reduce.asp